lib/tbot/tbot.go

```diff
@@ -68,6 +68,11 @@ func (b *Bot) setClient(client auth.ClientI) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	// Make sure the previous client is closed.
+	if b._client != nil {
+		defer b._client.Close()
```

Comment:
Any reason to defer this? I'm pretty sure it does the right thing but it's a bit less clear which client is being closed

Reply:
That's fair, it's unnecessarily confusing for no reason. I've removed the `defer`!